healthcare/api-client/v1/dicom/dicomweb_test.py

```diff
@@ -0,0 +1,249 @@
+    def create():
+        try:
+            dicom_stores.create_dicom_store(
+                project_id, cloud_region, dataset_id, dicom_store_id
+            )
+        except HttpError as err:
+            # We ignore 409 conflict here, because we know it's most
+            # likely the first request failed on the client side, but
+            # the creation suceeded on the server side.
+            if err.resp.status == 409:
+                print(
+                    "Got exception {} while creating DICOM store".format(
+                        err.resp.status
+                    )
+                )
+            else:
+                raise
+
+    create()
+
+    yield
+
+    # Clean up
+    @backoff.on_exception(backoff.expo, HttpError, max_time=60)
+    def clean_up():
+        try:
+            dicom_stores.delete_dicom_store(
+                project_id, cloud_region, dataset_id, dicom_store_id
+            )
+        except HttpError as err:
+            # The API returns 403 when the DICOM store doesn't exist.
+            if err.resp.status == 404 or err.resp.status == 403:
+                print(
+                    "Got exception {} while deleting DICOM store".format(
+                        err.resp.status
+                    )
+                )
+            else:
+                raise
+
+    clean_up()
+
+
+def test_dicomweb_store_instance(test_dataset, test_dicom_store, capsys):
+    dicomweb.dicomweb_store_instance(
+        base_url, project_id, cloud_region, dataset_id, dicom_store_id, dcm_file
+    )
+
+    out, _ = capsys.readouterr()
+
+    # Check that store instance worked
+    assert "Stored DICOM instance" in out
+
+
+def test_dicomweb_search_instance_studies(test_dataset, test_dicom_store, capsys):
+    dicomweb.dicomweb_store_instance(
+        base_url, project_id, cloud_region, dataset_id, dicom_store_id, dcm_file
+    )
+
+    dicomweb.dicomweb_search_instance(
+        base_url, project_id, cloud_region, dataset_id, dicom_store_id
+    )
+
+    dicomweb.dicomweb_search_studies(
+        base_url, project_id, cloud_region, dataset_id, dicom_store_id
+    )
+
+    out, _ = capsys.readouterr()
+
+    # Check that search instance worked
+    assert "Instances:" in out
+    # Check that search studies worked
+    assert "Studies found: response is <Response [204]>" in out
+
+
+def test_dicomweb_retrieve_study(test_dataset, test_dicom_store, capsys):
+    dicomweb.dicomweb_store_instance(
+        base_url, project_id, cloud_region, dataset_id, dicom_store_id, dcm_file
+    )
+
+    dicomweb.dicomweb_retrieve_study(
+        base_url, project_id, cloud_region, dataset_id, dicom_store_id, study_uid
+    )
+
+    # Assert study was downloaded
+    assert os.path.isfile("study.multipart")
+
+    out, _ = capsys.readouterr()
+
+    # Check that retrieve study worked
+    assert "Retrieved study" in out
+
+    # Delete downloaded study
+    os.remove("study.multipart")
```

Comment:
Not a big deal, but can you put the `os.remove()` call in `finally` clause?

```python
    try:
        ...
        ...
      finally:
        os.remove("study.multipart")
```

also for some tests below

Reply:
Done.